Z3/smtlib-backends-z3.cabal

```diff
@@ -28,7 +28,7 @@ library
   ghc-options:      -Wall -Wunused-packages
   exposed-modules:  SMTLIB.Backends.Z3
   build-depends:
-      base             >=4.15 && <4.17.0
+      base             >=4.14 && <4.17.0
```

Comment:
Why is this change necessary ? And if it is not, how did you know you could relax the lower bound on the version of `base` ? I used `4.15` because that's what auto-complete gave me but I wonder if there's a way to know up until which version is a piece of code valid.